test/real_content/test_swupd.bats

```diff
@@ -0,0 +1,211 @@
+
+# get_format: Echos the format of the version in $1
+# TODO: This is not the best way to handle that. we should have in swupd
+# codebase the information about the supported format of this swupd.
+get_format() {
+	v="$1"
+	curl -f ${URL}/${v}/format 2>/dev/null
+	return 0
+}
+
+# get_format_from_versions: Echos the first format available in a version from
+# a list of versions
+get_format_from_versions() {
+	for v in "$@"; do
+		format=$(get_format $v)
+		if [ -z "$format" ]; then # if version isn't published, discard
+			continue
+		else
+			echo $format
+			break
+		fi
+	done
+}
+
+test_setup() {
+	# Fill defaults in optional parameters
+	if [ -z "$URL" ]; then
+		URL=https://cdn.download.clearlinux.org/update
+	fi
+	if [ -z "$MAX_VERSIONS" ]; then
+		MAX_VERSIONS=15
+	fi
+
+	# Using github clr-bundles project to get official clear releases
+	#TODO: Discover available versions to work with custom mixers
+	#Depends on: https://github.com/clearlinux/mixer-tools/issues/467
+	version_list=$(git ls-remote --tags https://github.com/clearlinux/clr-bundles.git | grep -v {} | grep -v latest | sed "s/.*refs\/tags\///" | sort -gr | head -n $MAX_VERSIONS)
+	if [ -z "$version_list" ]; then
+		echo "Impossible to get versions list"
+		return 1
+
+	fi
+
+	# Get the last format
+	export FORMAT=$(get_format_from_versions $version_list)
+	version_list=$(echo "$version_list" | sort -g)
+	if [ -z "$FORMAT" ]; then
+		echo "Impossible to guess format"
+		return 1
+
+	fi
+
+	# Fill a list of last available versions (up to ${MAX_VERSIONS})
+	i=0
+	for v in $version_list; do
+		format=$(get_format $v)
+		# if version isn't published or is from a different version, discard
+		if [ -z "$format" ] || [ "$FORMAT" -ne "${format}" ]; then
+			continue
+		fi
+
+		VERSION[$i]=$v
+		i=$(($i+1))
+	done
+
+	# Not enough versions to run the test
+	if [ -z "${VERSION[0]}" ] || [ -z "${VERSION[1]}" ]; then
+		#TODO: This is going to break on first release in a new format
+		echo "We need at least 2 versions in format $FORMAT to continue with this test"
+		return 1
+
+	fi
+
+	sudo rm -rf "$TEST_NAME"
+	sudo mkdir -p "$TEST_NAME"
+	ROOT_DIR="${PWD}/${TEST_NAME}"
+
+	SWUPD_OPTS="-u ${URL} -p ${ROOT_DIR} -S ${ROOT_DIR}/var/lib/swupd/"
+}
+
+test_teardown() {
+	if [ "$DEBUG_TEST" != true ]; then
+		sudo rm -rf "$TEST_NAME"
+	fi
+	return
+}
+
+verify_system() {
+	#TODO: FIX
+	# There are validation problems happening in diverse files. Keeping this disabled until we
+	# have this fixed in verify fix.
+
+	#run sudo sh -c "$SWUPD verify --picky $SWUPD_OPTS 2>/dev/null"
+	#if [ ! -z "$output" ]; then
+	#	echo "Verify found extra files in the system:"
+	#	echo "$output"
+	#	return 1
+	#fi
+
+	#run sudo sh -c "$SWUPD verify $SWUPD_OPTS"
+	#assert_not_in_output "did not match"
+
+	return
+}
+
+# check_version: Check if version in /usr/lib/os-release is equal $1.
+# Fails otherwise.
+check_version() {
+	version="$1"
+	cur_version=$(grep VERSION_ID ${ROOT_DIR}/usr/lib/os-release|cut -d = -f 2)
```

Comment:
optional: I like using AWK instead:

```shell
cur_version=$(awk -F= '$1 == "VERSION_ID" { print $2 }' ${ROOT_DIR}/usr/lib/os-release)
```

Reply:
For this case I see no reason to use awk, but I know shell-script devs usually hate grep|cut

Reply:
I'm okay with this change, I just prefer the other way. One improvement would be to grep for `^VERSION_ID=` to avoid matching some other key (or value).